src/shared/libfido2-util.c

```diff
@@ -520,6 +636,32 @@ int fido2_use_hmac_hash(
                         goto finish;
                 }
 
+                bool in_token = false;
+
+                r = fido2_is_cred_in_specific_token(path, rp_id, cid, cid_size, pins, required, &in_token);
+                /* We handle PIN related errors here since we have used PIN in the check.
+                 * If PIN error happens, we can avoid pin retries decreased the second time. */
+                if (r < 0) {
+                        if (IN_SET(r, -ENOANO, -ENOLCK, -EOWNERDEAD)) {
+                                /* If it's not the last device, try next one */
+                                if (i < found - 1)
+                                        continue;
+                                else {
+                                        if (r == -EOWNERDEAD)
+                                                r = -EAGAIN;
```

Comment:
what's the rationale for this?

Reply:
This make it behave same as https://github.com/systemd/systemd/blob/main/src/shared/libfido2-util.c#L529

Reply:
there should be a comment